app/scripts/controllers/network/network.js

```diff
@@ -418,8 +418,7 @@ export default class NetworkController extends EventEmitter {
 
   _configureStandardProvider(rpcUrl, chainId) {
     log.info('NetworkController - configureStandardProvider', rpcUrl);
-    const networkClient = createJsonRpcClient({ rpcUrl, chainId });
-    this._setNetworkClient(networkClient);
+    createJsonRpcClient({ rpcUrl, chainId }).then(this._setNetworkClient);
```

Comment:
It's more idiomatic (and fits with our style) to use `async`/`await` syntax here instead of `.then`. That said, making this method asynchronous is going to make everything up the call stack asynchronous as well. That makes this change a lot trickier than I first thought and indicates that this might not be the right approach for this 🤔

Reply:
I tried using `createAsyncMiddleware` in 11da0a95c55f46e673fc80cfaeabc61cf0646036. Probably did it wrong but is that a workable solution?

Reply:
Unfortunately, I don't think that's going to work. You can't use middleware to build other middleware — it's expected that you'll do something with a request and response (see example [here](https://github.com/MetaMask/json-rpc-engine#async-middleware)). This will require some more thought.

Reply:
Ah got it, back to the drawing board I guess 😅